addons/mrp/models/mrp_workorder.py

```diff
@@ -159,6 +160,19 @@ def _onchange_finished_lot_id(self):
             if line:
                 self.qty_producing = line.qty_done
 
+    @api.onchange('date_planned_start', 'date_planned_finished')
+    def _onchange_planned_dates(self):
+        if self.date_planned_start and self.date_planned_finished:
+            diff = self.date_planned_finished - self.date_planned_start
+            self.duration_expected = diff.total_seconds() / 60
```

Comment:
won't this trigger the other onchange, reseting date planned finished? maybe you need to put all of this in the same onchange

Reply:
@sle-odoo, Yes it will call other onchange but that way it will sync perfectly. If we put all code in one onchage then it will be called twice(Maybe we have find which field is changed).
Something like:
```
@api.onchange('date_planned_start', 'date_planned_finished', 'duration_expected')
def _onchange_planned_dates(self):
    if self.date_planned_start or self.date_planned_finished is changed?:
        # By assigning value to duration it will call this method again as 'duration_expected' field is in trigger list
        self.duration_expected = some stuff
    elif self.duration_expected is changed?:
        # same as above
        self.date_planned_finished = some stuff
```